cloudsmith/data_entitlement.go

```diff
@@ -0,0 +1,357 @@
+		if err != nil {
+			return nil, err
+		}
+		tokensList = append(tokensList, tokensPage...)
+		pageCurrentCount++
+	}
+
+	return tokensList, nil
+}
+
+func flattenEntitlementToken(token []cloudsmith.RepositoryToken) []interface{} {
+	tokens := make([]interface{}, len(token))
+
+	for i, t := range token {
+		log.Printf("[DEBUG] Entitlement Token: %v", t)
+		token := make(map[string]interface{})
+		token["clients"] = t.GetClients()
+		token["created_at"] = t.GetCreatedAt().Format(time.RFC3339)
+		token["created_by"] = t.GetCreatedBy()
+		token["default"] = t.GetDefault()
+		token["downloads"] = t.GetDownloads()
+		token["disable_url"] = t.GetDisableUrl()
+		token["enable_url"] = t.GetEnableUrl()
+		token["eula_required"] = t.GetEulaRequired()
+		token["has_limits"] = t.GetHasLimits()
+		token["identifier"] = t.GetIdentifier()
+		token["is_active"] = t.GetIsActive()
+		token["is_limited"] = t.GetIsLimited()
+		token["limit_bandwidth"] = t.GetLimitBandwidth()
+		token["limit_bandwidth_unit"] = t.GetLimitBandwidthUnit()
+		token["limit_date_range_from"] = t.GetLimitDateRangeFrom().Format(time.RFC3339)
+		token["limit_date_range_to"] = t.GetLimitDateRangeTo().Format(time.RFC3339)
+		token["limit_num_clients"] = t.GetLimitNumClients()
+		token["limit_num_downloads"] = t.GetLimitNumDownloads()
+		token["limit_package_query"] = t.GetLimitPackageQuery()
+		token["limit_path_query"] = t.GetLimitPathQuery()
+		token["metadata"] = t.GetMetadata()
+		token["refresh_url"] = t.GetRefreshUrl()
+		token["reset_url"] = t.GetResetUrl()
+		token["scheduled_reset_at"] = t.GetScheduledResetAt().Format(time.RFC3339)
+		token["scheduled_reset_period"] = t.GetScheduledResetPeriod()
+		token["self_url"] = t.GetSelfUrl()
+		token["slug_perm"] = t.GetSlugPerm()
+		token["token"] = t.GetToken()
+		token["updated_at"] = t.GetUpdatedAt().Format(time.RFC3339)
+		token["updated_by"] = t.GetUpdatedBy()
+		token["updated_by_url"] = t.GetUpdatedByUrl()
+		token["usage"] = t.GetUsage()
+		token["user"] = t.GetUser()
+		token["user_url"] = t.GetUserUrl()
+
+		tokens[i] = token
+	}
+
+	return tokens
+}
+
+func dataSourceEntitlementRead(d *schema.ResourceData, m interface{}) error {
+	pc := m.(*providerConfig)
+
+	namespace := requiredString(d, "namespace")
+	repository := requiredString(d, "repository")
+	query := buildQueryString(d.Get("query").(*schema.Set))
+	showTokenVal := optionalBool(d, "show_token")
+	activeTokenVal := optionalBool(d, "active_token")
+
+	var pageCount, pageSize int64 = -1, -1
+
+	entitlementList, err := retrieveEntitlmentListPages(pc, namespace, repository, query, pageSize, pageCount, *showTokenVal, *activeTokenVal)
+	if err != nil {
+		return err
+	}
+
+	tokens := flattenEntitlementToken(entitlementList)
+	if err := d.Set("entitlement_tokens", tokens); err != nil {
+		return err
+	}
+
+	d.SetId(strconv.FormatInt(time.Now().Unix(), 10))
```

Comment:
❓ Question for my understanding. Is setting this ID here is to avoid terraform retrieving the data on the next run?